### DISCUSSION TITLE
Discussion about tests

### DISCUSSION THREAD

**Discussion by rubenquadros** [hieuwu/android-groceries-store#213] [General]
Hey @hieuwu!

Just a few questions about the sample [use case test](https://github.com/hieuwu/android-groceries-store/blob/main/app/src/test/java/com/hieuwu/groceriesstore/domain/usecases/impl/GetProfileUseCaseImplTest.kt)
 
-  Instead of mocking the repository should we create a fake?
- Instead of using `runBlocking` should we use [runTest](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-test/kotlinx.coroutines.test/run-test.html)?  More info [here](https://developer.android.com/kotlin/coroutines/test)
- Should we consider using [turbine](https://github.com/cashapp/turbine) to test flows?

**Comment by hieuwu** [comment]
- In terms of mocking repository, I thought about creating a fake before as official document from Android team states. However, it seems unecessary. If we create a fake for each repository, we have to implement a mock version (fake) for it and there is no way to make sure the implementation of the mock is the same with tested one. It’s quite hard to review and ensure the result of the repository
- Yes. Use runTest is better.
- Haven’t try it yet. Could you please share more about it?

**Reply by rubenquadros** [reply]
The github turbine repo has lot of information. You can also checkout this [article](https://code.cash.app/flow-testing-with-turbine) for some more info.

**Reply by hieuwu** [reply]
Great. Your feedback is really helpful. I''l have a plan to apply it to improve the repo. Thanks a lot

**Comment by hieuwu** [comment]
> The github turbine repo has lot of information. You can also checkout this [article](https://code.cash.app/flow-testing-with-turbine) for some more info.
